spec/block_category.spec.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+use Kahlan\Arg;
+
+describe(\GovukComponents\BlockCategory::class, function () {
+    beforeEach(function () {
+        $this->blockCategory = new \GovukComponents\BlockCategory();
+    });
+
+    describe('->register()', function () {
+        it('adds the filter', function () {
+            allow('add_filter')->toBeCalled();
+            expect('add_filter')->toBeCalled()->once()->with('block_categories', [$this->blockCategory, 'newBlocksCategory'], 10, 1);
+            $this->blockCategory->register();
+        });
+    });
+
+    describe('->newBlocksCategory()', function () {
+        it('appends a new value to the array, and returns it', function () {
```

Comment:
There's a second path through this method, which is currently untested: if the $categories array already has a category with the `govuk-custom` slug, it returns the array unamended. We should test for that context too, for completeness.

Reply:
1. I've added a comment at the top, could you please let me know if that's ok please? Thank you

Reply:
That's a really good summary of why we've made the change, plus an extra bit of context for how we might want to expand on it in future 🙂, nice work! The only thing I'd suggest is modifying the commit message to include that in text the body, as well as putting at the top of the PR. That way it'd be right at the fingertips of someone looking through the version history in future.

Reply:
I'm not sure that this is correct way to modify the tests, bare with me.